### PR TITLE
.Environment.Name

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/rancher/rancher/issues/8791

### CODE REVIEW

**Comment by joshwget** [comment]
Can you rebase this PR?

**Comment by joshwget** [comment]
There's a bit or function reordering here that I'd rather avoid. Can we try to make it work without those changes?

File: project/context.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Context struct {
 	ComposeFiles        []string
 	ComposeBytes        [][]byte
 	ProjectName         string
+	EnvName             string
```
Comment (joshwget) [inline project/context.go]: Minor nit, can you change `Env` to `Environment` here and other places where it's shortened?

File: project/project.go [hunk 2]
```diff
@@ -42,8 +42,8 @@ type Project struct {
 	hasListeners bool
 }
 
-// NewProject creates a new project with the specified context.
 func NewProject(context *Context) *Project {
+	// NewProject creates a new project with the specified context.
```
Comment (joshwget) [inline project/project.go]: Can you move this back? It'll cause golint to complain.

**Comment by joshwget** [comment]
LGTM

**Comment by cjellick** [comment]
@carolynhu can you add an integration test for this change please?

**Review by cjellick — CHANGES_REQUESTED** [review]
@carolynhu can you add an integration test for this change please?

**Comment by joshwget** [comment]
Test looks good.

**Comment by cjellick** [comment]
LGTM

**Review by cjellick — APPROVED** [review]

**Comment by cjellick** [comment]
Replaced this with a PR that was for master branch

**Pull request closed without merging**